### PR TITLE
New option --operator-csv-modifications-url

### PR DESCRIPTION
**Pull request by MartinBasti**
Option can pass URL to JSON file with modifications to operator CSV
file.

Can be used only with --isolated option

* CLOUDBLD-3976

Signed-off-by: Martin Bašti <mbasti@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [n/a] JSON/YAML configuration changes are updated in the relevant schema
- [n/a] Changes to metadata also update the documentation for the metadata
- [waived] Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

**Comment by ben-alkov** [comment]
@MartinBasti;
- Code coverage decreased slightly
- Should there be a docs PR?

**Comment by MartinBasti** [comment]
@ben-alkov I'll update docs when feature is completed, this is one of many PRs

File: osbs/cli/main.py [hunk 1]
```diff
@@ -733,6 +734,9 @@ def cli():
                               metavar="pkg_manager:name:version[:new_name]",
                               dest="dependency_replacements",
                               help="Cachito dependency replacement")
+    build_parser.add_argument("--operator-csv-modifications-url", action='store', required=False,
+                              dest='operator_csv_modifications_url', metavar='URL',
+                              help="URL to JSON file with operator CSV modification")
```
Comment (tkdchen) [inline osbs/cli/main.py:739]: It would be good to use ClusterServiceVersion in user-oriented message.

Comment (tkdchen) [reply osbs/cli/main.py:739]: Changed my mind. Not necessary to change it specifically.

**Comment by rcerven** [comment]
code looks good

**Comment by MartinBasti** [comment]
I don't know why coverage dropped only for centos. There are no unittests for CLI params

File: tests/test_api.py [hunk 2]
```diff
@@ -2650,6 +2656,80 @@ def __init__(self, isolated):
             assert isinstance(exc_info.value.cause, ValueError)
             assert "Flatpak build cannot be isolated" in exc_info.value.message
 
+    @pytest.mark.parametrize('isolated', [True, False])
+    def test_operator_csv_modifications_url_cli(self, osbs, isolated):
+        """Test if combination of isolated build and operator_csv_modifications_url
```
Comment (mkosiarc) [inline tests/test_api.py:2661]: This wording suggests to me that isolated build with operator csv modification url should raise an exception

Comment (lkolacek) [reply tests/test_api.py:2661]: +1

**Review by rcerven — APPROVED** [review]

File: tests/test_api.py (hunk 2 above)
Comment (MartinBasti) [reply tests/test_api.py]: Reworded

**Review by mkosiarc — APPROVED** [review]

**Comment by MartinBasti** [comment]
RN:
- New option `--operator-csv-modifications-url` for modifying operator bundle CSV file

**Pull request merged**